File: algorithms/neetcode/algorithms/algorithms/src/main/java/com/mycompany/leetcode/medium/FindTheScoreOfAllPrefixesOfAnArray/find-the-score-of-all-prefixes-of-an-array-rock.py

from typing import List

class S:
    def findPrefixScore(self, nums: List[int]) -> List[int]:
        score, mx = [0], 0
        for num in nums:
            mx = max(mx, num)
            score.append(score[-1] + num + num)
        return score[1:]

nums = [2,3,7,5,10]
print(S().findPrefixScore(nums))

File: algorithms/neetcode/algorithms/algorithms/src/main/java/com/mycompany/leetcode/medium/FindTheScoreOfAllPrefixesOfAnArray/FindTheScoreOfAllPrefixesOfAnArrayRock.java
package com.mycompany.leetcode.medium.FindTheScoreOfAllPrefixesOfAnArray;

import java.util.Arrays;

public class FindTheScoreOfAllPrefixesOfAnArrayRock {
    static long[] findPrefixScore(int[] nums) {
        int n = nums.length;
        long[] score = new long[n + 1];
        for(int i = 0, max = 0; i < n; ++i) {
            max = Math.max(max, nums[i]);
            score[i + 1] += score[i] + max + nums[i];
        }
        return Arrays.copyOfRange(score, 1, n + 1);
    }

    public static void main(String[] args) {
        int[] nums = {2,3,7,5,10};
        long[] result = findPrefixScore(nums);
        System.out.println(Arrays.toString(result));
    }
}
